repository/repository.go

```diff
@@ -110,34 +112,93 @@ func OpenWithBasePath(ctx context.Context, repo string, basePath string) (*Repos
 		userRepoPath: userRepoPath,
 		forkRepoPath: forkRepoPath,
 		basePath:     expandedBasePath,
+		lockManager:  NewRepositoryLockManager(userRepoPath),
 	}
 
-	if err := r.ensureFork(ctx); err != nil {
-		return nil, fmt.Errorf("unable to fork the repository: %w", err)
-	}
-	if err := r.ensureUserRemote(ctx); err != nil {
-		return nil, fmt.Errorf("unable to set container-use remote: %w", err)
+	// Use repository-level lock for fork and remote setup to prevent concurrent conflicts
+	err = r.lockManager.WithLock(ctx, LockTypeRepo, func() error {
+		if err := r.ensureFork(ctx); err != nil {
+			return fmt.Errorf("unable to fork the repository: %w", err)
+		}
+		if err := r.ensureUserRemote(ctx); err != nil {
+			return fmt.Errorf("unable to set container-use remote: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
 	}
 
 	return r, nil
 }
 
 func (r *Repository) ensureFork(ctx context.Context) error {
-	// Make sure the fork repo path exists, otherwise create it
-	_, err := os.Stat(r.forkRepoPath)
-	if err == nil {
+	// Check if fork already exists
+	if _, err := os.Stat(r.forkRepoPath); err == nil {
 		return nil
+	} else if !os.IsNotExist(err) {
+		return err
+	}
+
+	// Create lock file in parent directory, named after the fork repo
+	lockDir := filepath.Dir(r.forkRepoPath)
+	repoName := filepath.Base(r.forkRepoPath)
+	lockFile := filepath.Join(lockDir, repoName+".lock")
+
+	// Ensure lock directory exists
+	if err := os.MkdirAll(lockDir, 0755); err != nil {
+		return fmt.Errorf("failed to create lock directory: %w", err)
 	}
```

Comment:
🐛 **Bug (Medium):** Magic number should be a named constant, and traditional for loop is clearer for retry logic

```suggestion
const maxForkRetries = 10
	for i := 0; i < maxForkRetries; i++ {
```